tests/e2e/internal/retrieval/test_series.py

```diff
@@ -66,113 +122,385 @@ def test_fetch_series_values_single_series(
-    expected_end,
 ):
     # given
-    run_definition = RunAttributeDefinition(experiment_identifier, AttributeDefinition(attribute_name, attribute_type))
+    sys_id = get_sys_id_for_run(client, project_1.project_identifier, project_1.ingested_runs[0].run_id)
+    run_definition = RunAttributeDefinition(
+        RunIdentifier(project_1.project_identifier, sys_id),
+        AttributeDefinition(attribute_name, attribute_type),
+    )
 
-    #  when
+    # when
     series = extract_pages(
         fetch_series_values(
             client,
             [run_definition],
             include_inherited=False,
-            container_type=ContainerType.EXPERIMENT,
-            step_range=step_range,
+            container_type=ContainerType.RUN,
+            **additional_kws,
         )
     )
 
     # then
-    expected = [
-        (step, value, int((NOW + timedelta(seconds=int(step))).timestamp()) * 1000)
-        for step, value in list(enumerate(expected_values))[expected_start:expected_end]
-    ]
-    if expected:
-        assert series == [(run_definition, expected)]
-    else:
+    if not expected_values:
         assert series == []
+    else:
+        assert len(series) == 1
+        assert series[0][0] == run_definition
+        _, values = series[0]
 
+        # Fun fact: with tail_limit, the order of returned values is reversed
+        values = sorted(values)
+        assert len(values) == len(expected_values)
 
-@pytest.mark.parametrize(
-    "attribute_name, attribute_type, expected_values",
-    [
-        (STRING_SERIES_PATHS[0], "string_series", TEST_DATA.experiments[0].string_series[STRING_SERIES_PATHS[0]]),
-        (STRING_SERIES_PATHS[1], "string_series", TEST_DATA.experiments[0].string_series[STRING_SERIES_PATHS[1]]),
-        (
-            HISTOGRAM_SERIES_PATHS[0],
-            "histogram_series",
-            TEST_DATA.experiments[0].fetcher_histogram_series()[HISTOGRAM_SERIES_PATHS[0]],
-        ),
-        (FILE_SERIES_PATHS[0], "file_series", TEST_DATA.experiments[0].file_series_matchers()[FILE_SERIES_PATHS[0]]),
-    ],
-)
-@pytest.mark.parametrize(
-    "tail_limit",
-    [0, 1, 5, 20, 40],
-)
-def test_fetch_series_values_single_series_tail_limit(
-    client, project, experiment_identifier, attribute_name, attribute_type, expected_values, tail_limit
-):
-    # given
-    run_definition = RunAttributeDefinition(experiment_identifier, AttributeDefinition(attribute_name, attribute_type))
+        for i, (expected_step, expected_value) in enumerate(expected_values):
+            (step, value, timestamp) = values[i]
 
-    #  when
-    series = extract_pages(
-        fetch_series_values(
-            client,
-            [run_definition],
-            include_inherited=False,
-            container_type=ContainerType.EXPERIMENT,
-            tail_limit=tail_limit,
-        )
-    )
+            assert step == expected_step
+            assert value == expected_value
+            # Don't check the timestamp
```

Comment:
Please make it a TODO, at least. It should be checked and I'm not sure why we're not doing it now.

Reply:
adding the timestamp check